cdap-hbase-compat-base/src/test/java/co/cask/cdap/data2/util/ReferenceCountedSupplierTests.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.util;
+
+import co.cask.cdap.data2.transaction.coprocessor.CConfigurationCacheSupplier;
+import co.cask.cdap.data2.transaction.coprocessor.CacheSupplier;
+import co.cask.cdap.messaging.TopicMetadataCacheSupplier;
+import com.google.common.collect.Lists;
+import com.google.common.util.concurrent.Service;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Tests {@link ReferenceCountedSupplier}s.
+ */
+public class ReferenceCountedSupplierTests {
```

Comment:
I see this exception when I run this test -
```
Exception in thread "cdap-configuration-cache-refresh" java.lang.NullPointerException
	at org.apache.hadoop.hbase.client.HTable.backgroundFlushCommits(HTable.java:964)
	at org.apache.hadoop.hbase.client.HTable.flushCommits(HTable.java:1252)
	at org.apache.hadoop.hbase.client.HTable.close(HTable.java:1289)
	at co.cask.cdap.data2.util.hbase.ConfigurationTable.read(ConfigurationTable.java:152)
	at co.cask.cdap.data2.transaction.queue.hbase.coprocessor.CConfigurationReader.read(CConfigurationReader.java:39)
	at co.cask.cdap.data2.transaction.coprocessor.CConfigurationCache$1.run(CConfigurationCache.java:85)
Exception in thread "tms-topic-metadata-cache-refresh" java.lang.NullPointerException
	at co.cask.cdap.messaging.TopicMetadataCache.updateCache(TopicMetadataCache.java:123)
	at co.cask.cdap.messaging.TopicMetadataCache$2.run(TopicMetadataCache.java:183)
```

Is it because we use nulls as constructor arguments of `CConfigurationCacheSupplier` in the test?